compat-tests.el

```diff
@@ -2992,5 +2992,61 @@
   (with-temp-buffer
     (should-equal (take 3 (widget-create 'key)) '(key :value ""))))
 
+(ert-deftest compat-copy-tree ()
+  ;; Adapted from Emacs /test/lisp/subr-tests.el
+  ;; Check that values other than conses, vectors and records are
+  ;; neither copied nor traversed.
+  (cl-defstruct a foo)
+  (cl-defstruct c foo)
+  (cl-defstruct d)
+  (cl-defstruct e foo)
```

Comment:
Shouldn't these defstructs use a qualified name `compat-tests--a` in order to avoid namespace pollution?

Reply:
Yes, thank you.